### PR TITLE
volcano scheduler support custom request resource

### PR DESCRIPTION
**Pull request by Thor-wl**
- support custom resource request for volcano
- add basic e2e case for the feature

Signed-off-by: Thor <1187526662@qq.com>

### CODE REVIEW

**Comment by Thor-wl** [comment]
@googlebot I signed it!

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go [hunk 1]
```diff
@@ -286,6 +286,9 @@ type BatchSchedulerConfiguration struct {
 	// PriorityClassName stands for the name of k8s PriorityClass resource, it's being used in Volcano batch scheduler.
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
+	// CustomRequestResource stands for the resource list custom request for, it's being used in Volcano batch scheduler.
+	// +optional
+	CustomRequestResource apiv1.ResourceList `json:"customRequestResource,omitempty"`
```
Comment (liyinan926) [inline pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Optional field should have a pointer type.

**Comment by k82cn** [comment]
I'm not sure how this PR handle #964 :)

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: This is a map, so no pointer is ok

File: pkg/batchscheduler/volcano/volcano_scheduler.go [hunk 2]
```diff
@@ -75,7 +75,12 @@ func (v *VolcanoBatchScheduler) DoBatchSchedulingOnSubmission(app *v1beta2.Spark
 func (v *VolcanoBatchScheduler) syncPodGroupInClientMode(app *v1beta2.SparkApplication) error {
 	// We only care about the executor pods in client mode
 	if _, ok := app.Spec.Executor.Annotations[v1beta1.KubeGroupNameAnnotationKey]; !ok {
-		if err := v.syncPodGroup(app, 1, getExecutorRequestResource(app)); err == nil {
+		totalResource := getExecutorRequestResource(app)
+		customRequestResource := app.Spec.BatchSchedulerOptions.CustomRequestResource
+		if len(customRequestResource) > 0 {
+			totalResource = customRequestResource
+		}
```
Comment (hzxuzhonghu) [inline pkg/batchscheduler/volcano/volcano_scheduler.go:82]: if len(customRequestResource) > 0 {
			totalResource = customRequestResource
		} else{
		        totalResource := getExecutorRequestResource(app)
                }

File: pkg/batchscheduler/volcano/volcano_scheduler.go [hunk 3]
```diff
@@ -90,6 +95,10 @@ func (v *VolcanoBatchScheduler) syncPodGroupInClusterMode(app *v1beta2.SparkAppl
 	if _, ok := app.Spec.Driver.Annotations[v1beta1.KubeGroupNameAnnotationKey]; !ok {
 		//Both driver and executor resource will be considered.
 		totalResource := sumResourceList([]corev1.ResourceList{getExecutorRequestResource(app), getDriverRequestResource(app)})
```
Comment (william-wang) [inline pkg/batchscheduler/volcano/volcano_scheduler.go:97]: No need to calculate the totalResource if customRequestResource is configured. It's better move totalResource back a little bit.

Comment (Thor-wl) [reply pkg/batchscheduler/volcano/volcano_scheduler.go:99]: good advice. if customRequestResource is configured, it's no need to calculate getExecutorRequestResource again. You mean the same with @hzxuzhonghu 's  last advice

**Comment by Thor-wl** [comment]
> I'm not sure how this PR handle #964 :)

added filed CustomRequestResource in BatchSchedulerConfiguration provides a way to custom config resource reuqest.If configed in the yaml, the total request resource of exectors and drivers will not be calculated

File: pkg/batchscheduler/volcano/volcano_scheduler.go (hunk 2 above)
Comment (Thor-wl) [reply pkg/batchscheduler/volcano/volcano_scheduler.go]: done

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go (hunk 1 above)
Comment (liyinan926) [inline pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Can we rename this to simply `resources`?

Comment (william-wang) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Yes, we can use "minRes" or "minResource".

Comment (Thor-wl) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go]: done

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go [hunk 4]
```diff
@@ -286,6 +286,9 @@ type BatchSchedulerConfiguration struct {
 	// PriorityClassName stands for the name of k8s PriorityClass resource, it's being used in Volcano batch scheduler.
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
+	// MinResource stands for the resource list custom request for, it's being used in Volcano batch scheduler.
+	// +optional
+	MinResource apiv1.ResourceList `json:"minResource,omitempty"`
```
Comment (liyinan926) [inline pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Why it's named `MinResouce`? Does it define the lower limit of resource requests?

Comment (hzxuzhonghu) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Yes, this means the min resource request for gang scheduling

Comment (liyinan926) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: The comment should make it clear that this defines the lower-bound limit, and how it affects the resource request to the scheduler.

Comment (hzxuzhonghu) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: ping @Thor-wl

Comment (william-wang) [reply pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:291]: Is there any progress?

**Pull request merged**